src/main/java/courseManager/Outline.java

```diff
@@ -0,0 +1,98 @@
+package courseManager;
+
+import weightScheme.WeightScheme;
+
+import java.util.ArrayList;
+
+public class Outline {
+    private double runningGrade;
+    private double hypothethicalGrade;
+
+    private double percentageCompleted;
+    private int numberOfAssessmentsCompleted; /*do these need to be their own instance variables?*/
+    private ArrayList<Assessment> assessments;
+
+    public Outline () {
+        this.runningGrade = 0.0f;
+        this.hypothethicalGrade = 0.0f;
+        this.assessments = new ArrayList<>();
+    }
+
+    public void setRunningGrade(double runningGrade) {
+
+        this.runningGrade = runningGrade;
+    }
+
+    public void setHypothethicalGrade(double hypothethicalGrade) {
+
+        this.hypothethicalGrade = hypothethicalGrade;
+    }
+
+    public void addAssessment(Assessment assessment) {
+
+        assessments.add(assessment);
+    }
+
+    public void removeAssessment(Assessment assessment) {
+
+        assessments.remove(assessment);
+    }
+
+    public void editAssessment(int index, String title) {
+        assessments.get(index).setTitle(title);
+    }
+    public void editAssessment(int index, WeightScheme scheme) {
+        assessments.get(index).setWeightScheme(scheme);
+    }
+    public void editAssessment(int index, String title, WeightScheme scheme) {
+        assessments.get(index).setTitle(title);
+        assessments.get(index).setWeightScheme(scheme);
+    }
+
+    public Assessment getAssessmentData(int index) {
+        return assessments.get(index);
+    }
+
+    public double getPercentageCompleted() {
+        double percentageCompleted = 0.0f;
+        for (Assessment assessment : assessments) {
+            percentageCompleted += assessment.getSubmittedWeight();
+        }
+        return percentageCompleted;
+    }
+
+    public int getNumberOfAssessmentsCompleted() {
+        int numberOfAssessmentsCompleted = 0;
+        for (Assessment assessment : assessments) {
+            numberOfAssessmentsCompleted += assessment.getInstanceList().getNumberOfSubmittedInstances();
+        }
+        return numberOfAssessmentsCompleted;
+    }
+
+    public void computeRunningGrade() {
+        double rgrade = 0.0;
+        for (Assessment assessment : assessments) {
+            rgrade += assessment.getWeightScheme().computeWeighted(assessment.getInstanceList().getCommittedMarks());
```

Comment:
This logic should be handled by the assessment itself. consider using the getCommittedWeight function or if that's not enough add a function that does this.

Reply:
i overhauled this a little bit, let me know once I do the new pull request